lib/rules/no-get-by-assert.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const { getDocsUrl } = require('../utils');
+
+const isDirectlyCalledByFunction = node =>
+  node.parent.parent.type === 'CallExpression';
+
+const isReturnedByArrowFunctionExpression = node =>
+  node.parent.parent.type === 'ArrowFunctionExpression';
+
+const isDeclared = node => node.parent.parent.type === 'VariableDeclarator';
+
+const isReturnedByReturnStatement = node =>
+  node.parent.parent.type === 'ReturnStatement';
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Disallow `getBy*` queries as assertion method itself',
+      category: 'Best Practices',
+      recommended: false,
+      url: getDocsUrl('no-get-by-assert'),
+    },
+    messages: {
+      noGetByAssert: 'Disallowed use of `getBy*` query as implicit assert',
+    },
+    fixable: null,
+    schema: [],
+  },
+
+  create: function(context) {
+    return {
+      [`CallExpression > Identifier[name=${/^getBy(LabelText|PlaceholderText|Text|AltText|Title|DisplayValue|Role|TestId)$/}]`](
```

Comment:
Could this be extendable by the user if needed? For example we have some custom selectors e.g. `getByIcon` that I would like to fall under the same rule but should not be part of the default configuration.

Reply:
Yes, I can make it extendable so users can add their own queries. I always have doubts about just checking `getBy*` default queries + custom ones or check everything starting by `getBy`, but I think I prefer the first one and make it extendable so you and other users can add custom queries and avoid false positives with other things.